commands/parse.go

```diff
@@ -128,8 +136,42 @@ func createWebServer(quit chan<- string) *http.Server {
 	}
 }
 
-func waitForShutDown(server *http.Server, done chan<- string, quit <-chan string) {
-	content := <-quit
+func waitForShutDown(server *http.Server, done chan<- string, quit <-chan string, problemsCount int) {
+
+	readProblems := func() <-chan string {
```

Comment:
Can you add an inline comment as to where we have two function declaration inside the main `waitForShutDown` func?